kqueen/blueprints/api/views.py

```diff
@@ -62,6 +62,13 @@ def index():
 class ListClusters(ListView):
     object_class = Cluster
 
+    def get_content(self, *args, **kwargs):
+        clusters = super(ListClusters, self).get_content(*args, **kwargs)
+        # refresh states on all clusters
+        # TODO: replace this with some async function
+        [c.get_state() for c in clusters]
```

Comment:
Do we really want to run this on all clusters? I'm afraid this call can be really slow.

Reply:
It slows down the overview page loading, but its the only way to get the page to update the states correctly at the momment

Reply:
Understand ... I'l try to use some kind of background threading for this call since we probably don't need synchronous update.

Reply:
It isn't possible to use threading because the we'll get out of appication context. I'll try to do more hacking about it but we may stick with your solution as it may be the only way to go.